src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1878,4 +1880,79 @@ abstract class MergeIntoSuiteBase
       (1, 10) // update
     )
   )
+
+  protected def withJobCountListener(jobCount: AtomicInteger)(f: => Unit): Unit = {
```

Comment:
This is great, but it can be even cleaner. You can directly return the count as an int rather than having the caller provide the atomic integer. So basically 
```
protected def withJobCountListener(f: => Unit): Int = {
  val jobCount = new AtomicInteger
   ...
  jobCount.get()
}
```